### DISCUSSION TITLE
calculation with spin-orbit coupling of polar material

### DISCUSSION THREAD

**Discussion by ZhichaoLiu-0306** [nakib/elphbolt#76] [Q&A] [answered]
Dear Nakib,

The input file of 2nd Force constants is "espresso.ifc2", but for SOC calculation in EPW needed is espresso.ifc2.xml which cannot be read by src/phonon.f90 in elphbolt. Can the current version be used for SOC calculation?

If calculating polar materials, should I add "lpolar = .true." and "lifc  = .true." in "epw.in" to read "Read Z* and epsilon", or just make NAC correction in elphbolt rather than in epw?

Thanks!

**Comment by nakib** [comment, marked as answer]
Hi Zhichao

Thank you for your question!

In this case you can run the phonon calculation twice. Once generating the ifc2 file in the "normal" format to be passed to `elphbolt`. And another time to to generate it in the `xml` format to be used with the SOC enabled calculation with `EPW`. In the future I will consider implementing an `xml` format parser in `elphbolt`.

For polar materials, setting the flag `lpolar` in `epw.in` and `epsil` in `ph.in` to `.true.` will suffice. The NAC correction will be performed in `elphbolt` only if you set the flag `polar` to `.true.` in the `input.nml` file.

Make sure to pay attention to the key `spindeg` in `elphbolt` for your SOC enabled calculation.

I have not run any SOC enabled calculations, so I am curious to know how your calculations go. It is always a good idea to compare the electron and phonon bands and the e-ph matrix elements plotted along a path between `EPW` and `elphbolt`. Within numerical errors, they should be the same. I'd really appreciate it if you send feedback.

Best,
Nakib

**Reply by ZhichaoLiu-0306** [reply]
Dear Nakib,

Thanks a lot for your suggestions! I took a long time for the SOC calculation these days.

I have plotted the electron and phonon bands using the highsympath.txt. The results of elphbolt are consistent with those of EPW no matter whether the SOC is included. 

However, if I added "use_ws = .true." in EPW which could improve the quality of phonon dispersions compared with QE, elphbolt prints "No states found within Fermi window". How can I solve this problem？ 

Best,
Zhichao

**Reply by nakib** [reply]
Dear Zhichao,

Many thanks for your feedback. In the future I will add a few more material examples including at least one with the SOC enabled.

At the moment, I am not very familiar with the `use_ws` flag of `EPW`. The error message you shared emanates from the subroutine `apply_energy_window` and is displayed when no states are found in the specified Fermi window. If you share more specific information about when you encounter an issue with `elphbolt` when you have this flag turned on in EPW, I could try to take a look.

Best,
Nakib

**Reply by ZhichaoLiu-0306** [reply]
Dear nakib,

Thanks for your reply.

I have uploaded the output of test/Si with/without " use_ws " (https://github.com/ZhichaoLiu-0306/Si-test).  When using "use_ws", there is "No states found within Fermi window". But comment this line in epw.in, elphbolt runs all right. The difference in EPW is how to determine the Wigner-Seitz points shown in the line 395~396  of your modified ephwann_shuffle.f90.

The epmatwp1 are very big so they are left out in Si-test. If using "use_ws", cfac(d) and cfacq(d) will be calculated in the line 1158(1206) of modified ephwann_shuffle.f90 using ndegen_k(ir,iw2,iw) instead of ndegen_k(:,1,1).  I guess the format of elphbolt input files "wsdeg_*" might have to change if use_ws = .true. However, it may not need any modification since I am not familiar with EPW code and have a misunderstanding.  By the way, I just begin to learn EPW after reading your paper.

Best,
Zhichao

**Reply by nakib** [reply]
Hi Zhichao,

Thanks for digging deeper into the `use_ws` flag. I am traveling right now, but will take a closer look once I'm back. It would be great if you could also tell me what you see for the electronic band structure calculation with the `plot_along_path` flag when using `use_ws = .true.` in `EPW`.

Best,
Nakib

**Reply by ZhichaoLiu-0306** [reply]
Dear nakib,

Thanks! Wish you a pleasant journey!

![energy band](https://user-images.githubusercontent.com/76473203/206077887-b13792c6-3b98-47ef-9250-6490fd471336.jpg)
![phonon disperison](https://user-images.githubusercontent.com/76473203/206077906-65aefd15-d674-4d35-aba9-000900527e23.jpg)

The energy band is good using use_ws flag. For phonon dispersions, two TA branches from EPW and elphbolt are not degenerate along G-X-W with use_ws = .false. in fig2 (a), while using use_ws flag it is improved as shown in fig2(b).

Best,
Zhichao

**Reply by nakib** [reply]
Dear Zhichao,

Thanks a lot for sharing these. If possible, would you please share your `elphbolt` input.nml file?

I have created an issue (#77) and try to get to it soon, but, unfortunately, can't give a timeline.

Best regards,
Nakib

**Reply by ZhichaoLiu-0306** [reply]
Dear Nakib,

Thanks a lot !

The input.nml has been uploaded for SOC and no-SOC calculations (https://github.com/ZhichaoLiu-0306/Si-test).

Best regards,
Zhichao

**Reply by nakib** [reply]
Thanks!
